eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/Relayer.scala

```diff
@@ -148,51 +130,25 @@ class Relayer(nodeParams: NodeParams, router: ActorRef, register: ActorRef, paym
+      case _: Origin.LocalCold => postRestartCleaner ! r
+      case o: Origin.LocalHot => o.replyTo ! r
+      case _: Origin.ChannelRelayedCold => postRestartCleaner ! r
+      case _: Origin.ChannelRelayedHot => channelRelayer ! r
+      case _: Origin.TrampolineRelayedCold => postRestartCleaner ! r
+      case o: Origin.TrampolineRelayedHot => o.replyTo ! r
     }
 
-    case ff: ForwardFulfill => ff.to match {
-      case Origin.Local(_, None) => postRestartCleaner forward ff
-      case Origin.Local(_, Some(sender)) => sender ! ff
-      case Origin.Relayed(originChannelId, originHtlcId, amountIn, amountOut) =>
-        val cmd = CMD_FULFILL_HTLC(originHtlcId, ff.paymentPreimage, commit = true)
-        PendingRelayDb.safeSend(register, nodeParams.db.pendingRelay, originChannelId, cmd)
-        context.system.eventStream.publish(ChannelPaymentRelayed(amountIn, amountOut, ff.htlc.paymentHash, originChannelId, ff.htlc.channelId))
-      case Origin.TrampolineRelayed(_, None) => postRestartCleaner forward ff
-      case Origin.TrampolineRelayed(_, Some(paymentSender)) => paymentSender ! ff
-    }
-
-    case ff: ForwardFail => ff.to match {
-      case Origin.Local(_, None) => postRestartCleaner forward ff
-      case Origin.Local(_, Some(sender)) => sender ! ff
-      case Origin.Relayed(originChannelId, originHtlcId, _, _) =>
-        Metrics.recordPaymentRelayFailed(Tags.FailureType.Remote, Tags.RelayType.Channel)
-        val cmd = ff match {
-          case ForwardRemoteFail(fail, _, _) => CMD_FAIL_HTLC(originHtlcId, Left(fail.reason), commit = true)
-          case ForwardRemoteFailMalformed(fail, _, _) => CMD_FAIL_MALFORMED_HTLC(originHtlcId, fail.onionHash, fail.failureCode, commit = true)
-          case _: ForwardOnChainFail => CMD_FAIL_HTLC(originHtlcId, Right(PermanentChannelFailure), commit = true)
-        }
-        PendingRelayDb.safeSend(register, nodeParams.db.pendingRelay, originChannelId, cmd)
-      case Origin.TrampolineRelayed(_, None) => postRestartCleaner forward ff
-      case Origin.TrampolineRelayed(_, Some(paymentSender)) => paymentSender ! ff
-    }
-
-    case ChannelCommandResponse.Ok => () // ignoring responses from channels
+    case _: RES_SUCCESS[_] => () // ignoring responses from channels
 
     case GetChildActors(replyTo) => replyTo ! ChildActors(postRestartCleaner, channelRelayer, nodeRelayer)
   }
 
   override def mdc(currentMessage: Any): MDC = {
     val paymentHash_opt = currentMessage match {
-      case ForwardAdd(add, _) => Some(add.paymentHash)
-      case Status.Failure(addFailed: AddHtlcFailed) => Some(addFailed.paymentHash)
-      case ff: ForwardFulfill => Some(ff.htlc.paymentHash)
-      case ff: ForwardFail => Some(ff.htlc.paymentHash)
+      case RelayForward(add, _) => Some(add.paymentHash)
+      case addFailed: RES_ADD_FAILED[_] @unchecked => Some(addFailed.c.paymentHash)
+      case addCompleted: RES_ADD_COMPLETED[_, _] @unchecked => Some(addCompleted.htlc.paymentHash)
```

Comment:
I think the `@unchecked` aren't necessary.
nit2: there are unused imports in this file.

Reply:
Done with c46fff704.